lib/facter/kmod.rb

```diff
@@ -16,30 +16,28 @@
         }
 
         if File.directory?("/sys/module/#{directory}/parameters")
-          begin
-            Dir.foreach("/sys/module/#{directory}/parameters") do |param|
-              next if ['.', '..'].include?(param)
+          Dir.foreach("/sys/module/#{directory}/parameters") do |param|
+            next if ['.', '..'].include?(param)
 
-              next unless File.readable?("/sys/module/#{directory}/parameters/#{param}")
+            next unless File.readable?("/sys/module/#{directory}/parameters/#{param}")
 
+            begin
               kmod[directory]['parameters'][param] = File.read("/sys/module/#{directory}/parameters/#{param}").chomp
+            rescue StandardError => e
+              Facter.debug(e)
             end
-          rescue Errno::EACCES => e
-            Facter.debug(e)
-          rescue StandardError => e
-            Facter.warn(e)
           end
         end
 
         if File.directory?("/sys/module/#{directory}/holders")
-          begin
-            Dir.foreach("/sys/module/#{directory}/holders") do |used|
-              next if ['.', '..'].include?(used)
+          Dir.foreach("/sys/module/#{directory}/holders") do |used|
+            next if ['.', '..'].include?(used)
 
+            begin
               kmod[directory]['used_by'] << used
+            rescue StandardError => e
```

Comment:
This begin / rescue protect code that add a String in an Array… It should never fail :smile: 

I don't think listing holders could be a problem so maybe we can completely get rid of this one?

Reply:
:) took a stab at solving and pushed up.